src/plugins/data_views/public/services/has_data.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart, HttpStart } from 'kibana/public';
+import { DEFAULT_ASSETS_TO_IGNORE } from '../../common';
+import { HasDataViewsResponse, IndicesResponse, IndicesResponseModified } from '../';
+
+export class HasData {
+  private removeAliases = (source: IndicesResponseModified): boolean => !source.item.indices;
+
+  private isUserDataIndex = (source: IndicesResponseModified): boolean => {
+    // filter out indices that start with `.`
+    if (source.name.startsWith('.')) return false;
+
+    // filter out empty sources created by apm server
+    if (source.name.startsWith('apm-')) return false;
+
+    // filter out sources from FLEET_ASSETS_TO_IGNORE
+    for (const key in DEFAULT_ASSETS_TO_IGNORE) {
```

Comment:
I suggest to not do this dynamically like this because `DEFAULT_ASSETS_TO_IGNORE` have different type of resources, not only indices. 

also would avoid any

Reply:
done